packages/astro/src/runtime/server/render/component.ts

```diff
@@ -133,7 +133,14 @@ Did you mean to add ${formatList(probableRendererNames.map((r) => '`' + r + '`')
 		// If this component ran through `__astro_tag_component__`, we already know
 		// which renderer to match to and can skip the usual `check` calls.
 		// This will help us throw most relevant error message for modules with runtime errors
-		if (Component && (Component as any)[Renderer]) {
+		let isTagged = false;
+		try {
+			isTagged = Component && (Component as any)[Renderer];
+		} catch {
+			// Accessing `Component[Renderer]` may throw if `Component` is a Proxy that doesn't
+			// return the actual read-only value. In this case, ignore.
+		}
+		if (isTagged) {
```

Comment:
The error would be something like:

```
 error   'get' on proxy: property 'Symbol(astro:renderer)' is a read-only and non-configurable data property on the proxy target but the proxy did not return its actual value (expected '@astrojs/solid-js' but got 'props => {
      delete props.tag;
      return __vite_ssr_import_1__.createComponent(BaseComponent, __vite_ssr_imp...<omitted>... }')
```

Reply:
It would be very cheeky to parse the error message to get it's actual value now that I think of it, but right now it works without doing so.